code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -716,8 +729,9 @@ func (sc *StorageSmartContract) challengePassed(
 	if cab.success < cab.threshold {
 		partial = float64(cab.success) / float64(cab.threshold)
 	}
+	validators := getRandomSubSlice(cab.validators, validatorsRewarded, balances.GetBlock().GetRoundRandomSeed())
```

Comment:
We will only select 2 (`validators_per_challenge`) validators for each challenge, and we will pay reward to all of them, so adding an extra config might not be needed. Before we decide to increase the number of the validators in each challenge, we can leave it as it is now.

Reply:
The issue is that for each validator rewarded, we need to update their stake pool. So without this change, we need [validators_per_challenge](https://github.com/0chain/0chain/blob/staging/docker.local/config/sc.yaml#L153) MPT updates for each challenge.

If we only have two validators per challenge, that is probably not a big issue, but with ten validators, we could get a significant saving on smart contract timing. 

If we decided never to increase `validators_per_challenge` above two, we could argue we don't need this change.

Also, bear in mind https://github.com/0chain/0chain/issues/2505; we currently have no control over defective validators.

Reply:
Yes, 10 validators per challenge will lead to 10 times of validators stake pool updating, that's why we only select 2 of them. Adding this reward number may have limited effect on issue #2505, it would not help much until we increase the number.

Perhaps we can keep this changes in case we need to increase the validators number in for each challenge. But remember to make the number the same or less than the validtors_per_challenge.

Reply:
https://github.com/0chain/0chain/issues/2505 The simplest way to handle defective validators is to raise the number of validators per challenge so one bad one won't give an erroneous `challenge_response`. With two validators per challenge, any defective validator is causing an issue whenever it is used, so I think the number will be raised for any real chain.